src/atmos_flux_inversion/linalg.py

```diff
@@ -412,23 +429,30 @@ def _matmat(self, X):
         operator2 = self._operator2
         in_chunk = (operator1.shape[1], block_size, X.shape[1])
 
+        if isinstance(X, ARRAY_TYPES) and isinstance(operator1, ARRAY_TYPES):
+            partial_answer = einsum(
+                "ij,jkl->kil", operator1, X.reshape(in_chunk),
+                # Column-major output should speed the
+                # operator2 @ tmp bit
+                order="F"
+            ).reshape(block_size, -1)
```

Comment:
I think this is faster for NumPy/Dask arrays, but it may not be worth the code duplication, since `tensordot` also exists for those arrays.  It may also be worth looking into whether I can get this to work with `LinearOperator.matmul`, so I can use `LinearOperator`s for both slots in this Kronecker product operator as well.